File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.java
class Solution {
    public int maxChunksToSorted(int[] arr) {
        int max_so_far = arr[0], chunks = 0;
        for(int i=0; i<arr.length; i++){
            if(arr[i] > max_so_far){
                max_so_far = arr[i];
            }
            if(max_so_far == i){
                chunks++;
            }
        }
        return chunks;
    }
}

File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        stack = []
        for i in arr:
            if not stack or i >= stack[-1][-1]:
                stack.append([i])
            else:
                counter = 0
                j = len(stack) - 1
                while j >= 0:
                    if i < stack[j][-1]:
                        counter += 1
                        j -= 1
                    else:
                        break
                temp_list = [i]
                for k in range(counter):
                    temp_list.extend(stack.pop())
                stack.append(temp_list)
                stack[-1].sort()

        return len(stack)